### ISSUE TITLE
Create framework for hyperparameters optimization

### ISSUE THREAD

**Issue by andrefdre** [DanielCoelho112/AutoMec-AD#188]
After creating the cnn for driving #165 we need to find the best hyperparameters. The next step is to develop a script that test many configurations and then plots it's results.

**Comment by andrefdre** [comment]
I finished creating the script to test several hyperparameters to launch the script run:

```
rosrun prometheus_driving hyperparameter_optimization.py -d gazebo_right_lane -n_epoch 50 -batch_size 50 -c 0
```

Hyperparameters that will be tested:
- lr : [0.001, 0.0005 , 0.0001]
- lr_step_size : [10, 20,30,40]
- dropout : [0.1, 0.2, 0.3 ,0.4, 0.5]
- wd : [0.0001, 0.001, 0.01, 0.1]

Default hyperparameters:
- wd : 0
- dropout : 0
- lr : 0.001
- lr_step_size : 10

Do you think this are valid parameters?

This script will run several trainings and save them with the parameter name and its value. If you rerun the script, be sure to previously delete the created folders, else the script will prompt if you want to resume or overwrite the folder, so the script will be stuck waiting for input every time.

@manuelgitgomes After the meeting and discussion of the hyperparameters, could you run this script in your computer?

**Comment by andrefdre** [comment]
The networks were trained and these are the results:

Dropout:
![Dropout](https://github.com/AutoMecUA/AutoMec-AD/assets/58526188/1d0f0e4a-ded2-47bc-a8cf-c38cc6a5d09b)

Learning Rate:
![LR](https://github.com/AutoMecUA/AutoMec-AD/assets/58526188/1e3ebab5-1795-4d10-822a-f8ad723b374c)

Learning Rate Step Size:
![LR Step Size](https://github.com/AutoMecUA/AutoMec-AD/assets/58526188/196d15e1-6f47-453d-a0ef-7dc6a71992a1)

Weight Decay:
![wd](https://github.com/AutoMecUA/AutoMec-AD/assets/58526188/80776735-536e-49da-aae3-279e2076daa9)

**Comment by andrefdre** [comment]
This task is completed and the data was analyzed.
